### PR TITLE
Simplify windows version detection in vm

### PR DESCRIPTION
**Pull request by huoyaoyuan**
Uses the windows version helpers in SDK, which are pure header implemented.

### CODE REVIEW

File: src/coreclr/utilcode/util_nodependencies.cpp [hunk 1]
```diff
@@ -14,6 +14,11 @@
 #include "utilcode.h"
 #include "ex.h"
 
+#ifndef TARGET_UNIX
+#include <windows.h>
```
Comment (jkotas) [inline src/coreclr/utilcode/util_nodependencies.cpp:18]:
```suggestion
```
This should not be needed

**Review by jkotas — APPROVED** [review]
Thanks

File: src/coreclr/utilcode/util_nodependencies.cpp [hunk 2]
```diff
@@ -25,65 +29,26 @@ RunningOnStatusEnum gRunningOnStatus = RUNNING_ON_STATUS_UNINITED;
...
-        goto CHECK_SUPPORTED;
     }
-
-CHECK_SUPPORTED:
-
-    if (!fSupportedPlatform)
+    else
     {
         // The current platform isn't supported. Display a message to this effect and exit.
         fprintf(stderr, "Platform not supported: The minimum supported platform is Windows 7\n");
```
Comment (danmoseley) [inline src/coreclr/utilcode/util_nodependencies.cpp:48]: "Minimum supported version of Windows is"...?

Comment (jkotas) [inline src/coreclr/utilcode/util_nodependencies.cpp:48]:
```suggestion
        fprintf(stderr, "Platform not supported: Windows 7 is the minimum supported version\n");
```

**Comment by huoyaoyuan** [comment]
Is there any issue remaining?

**Pull request merged**